src/lib/features/project/project-read-model.test.ts

```diff
@@ -0,0 +1,162 @@
+});
+
+afterAll(async () => {
+    if (db) {
+        await db.destroy();
+    }
+});
+
+beforeEach(async () => {
+    await projectStore.deleteAll();
+    await flagStore.deleteAll();
+});
+
+test("it doesn't count flags multiple times when they have multiple events associated with them", async () => {
+    const projectId = 'project';
+    const flagName = 'flag';
+    await projectStore.create({ id: projectId, name: projectId });
+
+    await flagStore.create(projectId, { name: flagName, createdByUserId: 1 });
+
+    await eventStore.store({
+        type: 'feature-created',
+        createdBy: 'admin',
+        ip: '',
+        createdByUserId: 1,
+        featureName: flagName,
+        project: projectId,
+    });
+    await eventStore.store({
+        type: 'feature-updated',
+        createdBy: 'admin',
+        ip: '',
+        createdByUserId: 1,
+        featureName: flagName,
+        project: projectId,
+    });
+
+    const withFlags = await projectReadModel.getProjectsForAdminUi();
+    expect(withFlags).toEqual(
+        expect.arrayContaining([
+            expect.objectContaining({ id: projectId, featureCount: 1 }),
+        ]),
+    );
+});
+
+test('it uses the last flag change for an flag in the project as lastUpdated', async () => {
+    const projectId = 'project';
+    const flagName = 'flag';
+    await projectStore.create({ id: projectId, name: projectId });
+    await eventStore.store({
+        type: 'project-created',
+        createdBy: 'admin',
+        createdByUserId: 1,
+        project: projectId,
+        ip: '',
+    });
+
+    const noEvents = await projectReadModel.getProjectsForAdminUi();
+
+    expect(noEvents[0].lastUpdatedAt).toBeNull();
+
+    await flagStore.create(projectId, { name: flagName, createdByUserId: 1 });
+    await eventStore.store({
+        type: 'feature-created',
+        createdBy: 'admin',
+        ip: '',
+        createdByUserId: 1,
+        featureName: flagName,
+        project: projectId,
+    });
+
+    const withEvents = await projectReadModel.getProjectsForAdminUi();
+    expect(withEvents[0].lastUpdatedAt).not.toBeNull();
+});
+
+test('it does not consider flag events in a different project for lastUpdatedAt, and does not count flags belonging to a different project', async () => {
+    const projectId1 = 'project1';
+    await projectStore.create({ id: projectId1, name: 'Project1' });
+
+    const projectId2 = 'project2';
+    await projectStore.create({ id: projectId2, name: 'Project2' });
+
+    const flagName = 'flag';
+    await flagStore.create(projectId1, { name: flagName, createdByUserId: 1 });
+    await eventStore.store({
+        type: 'feature-created',
+        createdBy: 'admin',
+        ip: '',
+        createdByUserId: 1,
+        featureName: flagName,
+        project: projectId2,
+    });
+
+    const withEvents = await projectReadModel.getProjectsForAdminUi();
+
+    expect(withEvents).toEqual(
+        expect.arrayContaining([
```

Comment:
again I'd use toMatchObject if possible so that we can visually see the important parts of the object